ambry-router/src/test/java/com.github.ambry.router/GetOperationTest.java

```diff
@@ -0,0 +1,474 @@
+        networkClientFactory, new LoggingNotificationSystem(), mockClusterMap, time);
+    blobProperties = new BlobProperties(BLOB_SIZE, "serviceId", "memberId", "contentType", false, Utils.Infinite_Time);
+    userMetadata = new byte[BLOB_USER_METADATA_SIZE];
+    random.nextBytes(userMetadata);
+    putContent = new byte[BLOB_SIZE];
+    random.nextBytes(putContent);
+    ReadableStreamChannel putChannel = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(putContent));
+    blobIdStr = router.putBlob(blobProperties, userMetadata, putChannel).get();
+    router.close();
+    networkClient = networkClientFactory.getNetworkClient();
+  }
+
+  @Before
+  public void before() {
+    // almost all tests create and complete an operation outside of the router. Router's count of active operations
+    // will get decremented when that happens. Since the operation is not submitted via the router in this test,
+    // the count would never have been incremented. It is done here so that the count is ultimately zero when each test
+    // completes.
+    NonBlockingRouter.incrementOperationsCount();
+  }
+
+  @After
+  public void after() {
+    if (networkClient != null) {
+      networkClient.close();
+    }
+    Assert.assertEquals("All operations should have completed", 0, NonBlockingRouter.getOperationsCount());
+  }
+
+  /**
+   * Test {@link GetBlobInfoOperation} instantiation and validate the get methods.
+   * @throws Exception
+   */
+  @Test
+  public void testInstantiation()
+      throws Exception {
+    String blobIdStr = (new BlobId(mockClusterMap.getWritablePartitionIds().get(0))).getID();
+    // test a bad case
+    GetBlobInfoOperation op;
+    try {
+      new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, "invalid_id", operationFuture,
+          operationCallback, time);
+      Assert.fail("Instantiation of GetBlobInfo operation with an invalid blob id must fail");
+    } catch (RouterException e) {
+      Assert
+          .assertEquals("Unexpected exception received on creating GetBlobInfoOperation", RouterErrorCode.InvalidBlobId,
+              e.getErrorCode());
+    }
+
+    // test a good case
+    op = new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, blobIdStr, operationFuture,
+        operationCallback, time);
+
+    Assert.assertEquals("Callback must match", operationCallback, op.getCallback());
+    Assert.assertEquals("Futures must match", operationFuture, op.getFuture());
+    Assert.assertEquals("Blob ids must match", blobIdStr, op.getBlobIdStr());
+    // count was incremented, but this operation was not completed, so decrement.
+    NonBlockingRouter.decrementOperationsCount();
+  }
+
+  /**
+   * Test basic successful operation completion, by polling and handing over responses to the BlobInfo operation.
+   * @throws Exception
+   */
+  @Test
+  public void testPollAndResponseHandling()
+      throws Exception {
+    GetBlobInfoOperation op =
```

Comment:
suggestion: This seems to be attempting to test `GetOperation` by using `GetBlobInfoOperation`. This dependency might not be good (changes in `GetBlobInfoOperation` might require changes here, `GetBlobInfoOperation` bugs might fail tests in this class also, `GetBlobInfoOperation` might compensate/mask deficiencies in `GetOperation`, code in `GetOperation` might work only for `GetBlobInfoOperation` and not another child ). It's a little more effort but what do you think of having a test child class of `GetOperation`? That way you isolate only the code in `GetOperation`.